src/main.js

```diff
@@ -93,7 +94,16 @@ module.exports = {
       }
     }))
 
-    this.subscriptions.add(atom.commands.add('atom-text-editor', {
+    // Marks each JavaScript editor with a CSS class so that
+    // we can enable commands only for JavaScript editors.
+    this.subscriptions.add(atom.workspace.observeTextEditors((textEditor) => {
+      if (textEditor.getRootScopeDescriptor().scopes
```

Comment:
That makes sense and I agree. The issue I'm trying to address is when you have many linters installed, and you open the command pallet, start typing "fix" and then the fix commands for all your linters appear. Now you have the manual task of finding the right one.

The only way I know of how to scope commands is via CSS selectors (at least according to the [atom API](https://atom.io/docs/api/v1.15.0/CommandRegistry#instance-add)). With that in mind I see 2 options, either:

1.   Drop this PR and allow the command in all scopes
2.   Monitor each time the cursor moves, check the scope and add/remove the CSS selector
     - I don't like this option since it sounds pretty expensive and could slow the editor and ruin UX.

What are your feelings on this? Depending on your answser I will also amend my [TSLint PR](https://github.com/AtomLinter/linter-tslint/pull/149) accordingly.